src/lepton/lepton_codec.hh

```diff
@@ -112,6 +113,11 @@ protected:
         Sirikata::Array1d<std::vector<NeighborSummary>, (size_t)ColorChannel::NumBlockTypes> num_nonzeros_;
         uint32_t decode_index_;
         bool is_valid_range_;
+    public:
+        void init_bool_decoder(PacketReader* reader) {
```

Comment:
Let's code outside initialize and pass `std::unique_ptr<>` to `init_bool_decoder`?

Reply:
I made requested changes but not sure whether new code is more readable. Please check: https://github.com/Bulat-Ziganshin/lepton/commit/5168a2e37faa915f0059c24d81463c492402b9f4

Reply:
With modern C++ code like:
```cpp
std::unique_ptr<PacketReader> reader {new ActualThreadPacketReader(i, worker, send_to_actual_thread_state)};
ts->init_bool_decoder(std::move(reader));
```
will be replaced by:
```cpp
auto reader = std::make_unique<ActualThreadPacketReader>(i, worker, send_to_actual_thread_state);
ts->init_bool_decoder(std::move(reader));
```
Original comment wasn't for readability but for better memory control.

Reply:
std::make_unique available only in C++14 while the project is compiled with C++11 setting